### PR TITLE
Added keyword support for WithParser

### LINKED ISSUE #199: WithParser should have the ability to change keyword
Reinforce parsers should have constructor with keyword and priority

### PR DESCRIPTION
**Pull request by ikfir**
### ❓ Type of change

- [x] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (adds functionality)

### 📚 Description
Added keyword support for WithParser
Added new constructor to every `ArgumentParser` with those parameters `int, String`

Closes #199

### 🧪 How Has This Been Tested?
Unit test

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

## Release Notes for Version 0.0.36

- **New Features**
	- Enhanced `MinecraftVersion` support for versions 1.8 to 1.21.1 with protocol numbers.
	- Introduced new parsers: `UUIDParser`, `GameModeParser`, and various others for improved command handling.
	- Added `keyword` support in the `WithParser` annotation for better parser specification.

- **Bug Fixes**
	- Fixed protocol version retrieval to ensure server version checks.

- **Documentation**
	- Updated changelog and method documentation for clarity.

- **Refactor**
	- Improved constructor logic across multiple parser classes for maintainability.

### CODE REVIEW

File: commands/src/main/java/net/apartium/cocoabeans/commands/parsers/WithParserFactory.java [hunk 1]
```diff
@@ -36,35 +37,73 @@
...
             return List.of();
         }
     }
 
-    private static <T> T newInstance(Constructor<T>[] ctors, int priority) throws InstantiationException, IllegalAccessException, InvocationTargetException {
-        Constructor<?> constructor = null;
+    private static <T> T newInstance(Constructor<T>[] constructors, int priority, String keyword) throws InstantiationException, IllegalAccessException, InvocationTargetException {
+        Constructor<?> matchingConstructor = null;
+
+        boolean keywordBlank = keyword.isBlank();
```
[the remark this answers is not in the document]
Comment (ikfir) [reply commands/src/main/java/net/apartium/cocoabeans/commands/parsers/WithParserFactory.java:63]: Keyword couldn't be null annotations after all

File: commands/src/main/java/net/apartium/cocoabeans/commands/parsers/WithParserFactory.java [hunk 2]
```diff
@@ -36,35 +37,73 @@ public class WithParserFactory implements ParserFactory {
...
+        Class<?>[] paramTypes = null;
+        for (Constructor<T> constructor : constructors) {
+            int paramCount = constructor.getParameterCount();
+            paramTypes = constructor.getParameterTypes();
+
```
[the remark this answers is not in the document]
Comment (ikfir) [reply commands/src/main/java/net/apartium/cocoabeans/commands/parsers/WithParserFactory.java:68]: Maybe but that is only if someone change the code and doesn't read it and break it also unit test will catch it

File: commands/src/main/java/net/apartium/cocoabeans/commands/parsers/WithParserFactory.java [hunk 3]
```diff
@@ -36,35 +37,81 @@ public class WithParserFactory implements ParserFactory {
...
+            Constructor<? extends ArgumentParser<?>>[] ctors = (Constructor<? extends ArgumentParser<?>>[]) withParser.value().getConstructors();
+
+            ArgumentParser<?> argumentParser = newInstance(
+                    (Constructor<ArgumentParser<?>>[]) ctors,
+                    withParser.priority(),
+                    withParser.keyword()
+            );
+
+            if (argumentParser == null)
+                return List.of();
```
Comment (liorsl) [inline commands/src/main/java/net/apartium/cocoabeans/commands/parsers/WithParserFactory.java:49]: ....

Comment (ikfir) [reply commands/src/main/java/net/apartium/cocoabeans/commands/parsers/WithParserFactory.java:49]: Yeah what about it?
I don't return null just empty list

**Review by liorsl — APPROVED** [review]

**Pull request merged**